app/src/main/java/com/hedvig/app/feature/profile/ui/ProfileViewModelImpl.kt

```diff
@@ -57,23 +87,8 @@ class ProfileViewModelImpl(
         }
     }
 
-    override fun load() {
-        viewModelScope.launch {
-            profileRepository
-                .profile()
-                .onEach { response ->
-                    response.errors?.let {
-                        _data.value = ViewState.Error
-                        return@onEach
-                    }
-                    response.data?.let { _data.value = ViewState.Success(it) }
-                }
-                .catch { exception ->
-                    e(exception)
-                    _data.value = ViewState.Error
-                }
-                .launchIn(this)
-        }
+    override fun reload() {
+        observeProfileRetryChannel.retry()
```

Comment:
Changed this since as I see the old `load` function, if it were to be called twice (which it does if the load function is called which was called from the UI when the reload button was clicked), it'd actually launch 2 separate coroutines observing the `profile()`. This makes it only observe once, cancelling the previous collection if it was still active.

Reply:
Neat!